boefjes/boefjes/config.py

```diff
@@ -2,43 +2,58 @@
 from pathlib import Path
 from typing import Optional
 
-from pydantic import AnyHttpUrl, BaseSettings, PostgresDsn
+from pydantic import AmqpDsn, AnyHttpUrl, BaseSettings, Field, IPvAnyAddress, PostgresDsn
 
 
 class RuntimeConfiguration(Enum):
     LOCAL = "local"
 
 
 class Settings(BaseSettings):
-    base_dir: Path = Path(__file__).parent.resolve()
-    log_cfg: Path = Path(__file__).parent / "logging.json"
+    base_dir: Path = Field(Path(__file__).parent.resolve())
+    log_cfg: Path = Field(Path(__file__).parent / "logging.json", description="Path to the logging configuration file")
 
     # Worker configuration
-    pool_size: int = 2
-    poll_interval: float = 10.0
-    worker_heartbeat: float = 1.0
+    pool_size: int = Field(2, description="Number of workers to run per queue")
+    poll_interval: float = Field(10.0, description="Time to wait before polling for tasks when all queues are empty")
+    worker_heartbeat: float = Field(1.0, description="Seconds to wait before checking the workers when queues are full")
+
+    remote_ns: IPvAnyAddress = Field(
+        "1.1.1.1", description="Name server used for remote DNS resolution in the boefje runner"
+    )
 
     # Queue configuration
-    queue_uri: str = "amqp://"
+    queue_uri: AmqpDsn = Field(..., description="KAT queue URI", example="amqp://", env="QUEUE_URI")
 
     # Runtime configuration
-    runtime: RuntimeConfiguration = RuntimeConfiguration.LOCAL
-
-    katalogus_db_uri: PostgresDsn = "postgresql://xx:xx@host:5432/katalogus"
-
-    scheduler_api: AnyHttpUrl = "http://localhost:8004"
-
-    katalogus_api: AnyHttpUrl = "http://localhost:8003"
-
-    octopoes_api: AnyHttpUrl = "http://localhost:8001"
-
-    bytes_api: AnyHttpUrl = "http://localhost:8002"
-    bytes_username: str = "test"
-    bytes_password: str = "secret"
-
-    span_export_grpc_endpoint: Optional[str] = None
-
-    remote_ns: str = "1.1.1.1"
-
-
-settings = Settings()
```

Comment:
100% agree that this is a mediocre solution. The problem is that `settings-doc` attempts to load the module `config.py`, which results in a pydantic validation error given that `Settings()` is run. The only workaround that I can think of for this would be to explicitly provide some required environment variables when running `make docs`. But that's also not a clean solution. 

I haven't been able to find a way to cleanly inject environment variables, or how we can add an if statement to not load `Settings()` if run through `settings-doc`. Do you have any ideas for a solution? Now that I'm writing this, it occurs to me that an alright-ish middle ground could be to manually check (outside of Pydantic) if `CI=1` is set.

Reply:
I fixed it!